applications/StructuralMechanicsApplication/custom_elements/truss_elements/linear_truss_element_2D.cpp

```diff
@@ -0,0 +1,713 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:  Alejandro Cornejo
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "linear_truss_element_2D.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TNNodes>
+void LinearTrussElement2D<TNNodes>::Initialize(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Initialization should not be done again in a restart!
+    if (!rCurrentProcessInfo[IS_RESTARTED]) {
+        if (this->UseGeometryIntegrationMethod()) {
+            if (GetProperties().Has(INTEGRATION_ORDER) ) {
+                mThisIntegrationMethod = static_cast<GeometryData::IntegrationMethod>(GetProperties()[INTEGRATION_ORDER] - 1);
+            } else {
+                if constexpr (NNodes == 2) {
+                    mThisIntegrationMethod = GeometryData::IntegrationMethod::GI_GAUSS_1;
+                } else {
+                    mThisIntegrationMethod = GeometryData::IntegrationMethod::GI_GAUSS_2;
```

Comment:
this is duplicated several times, can you please refactor to a function?

Reply:
done